pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -141,17 +141,39 @@ type Webhook struct {
 	endpoint          string
 	resources         []string
 	operations        []admiv1.OperationType
-	filter            *containers.Filter
+	filters           map[string]*containers.Filter
 	containerRegistry string
 	pinnedLibraries   []libInfo
 }
 
 // NewWebhook returns a new Webhook
 func NewWebhook() (*Webhook, error) {
-	filter, err := apmSSINamespaceFilter()
+
+	var filters map[string]*containers.Filter = make(map[string]*containers.Filter);
+
+	ssiFilter, err := apmSSINamespaceFilter();
+	if err != nil {
+		return nil, err
+	}
+	filters["apm_config.instrumentation.enabled"] = ssiFilter;
+
+	asmFilter, err := apmProductNamespaceFilter("asm_config.enabled_namespaces", "asm_config.disabled_namespaces");
```

Comment:
ASM is not related to APM afaik (correct me if I am wrong).

Given this, I don't think the naming is good here (i.e. `apmProductNamespaceFilter(...)` is returning an `asm` filter)

Same applies to the following lines.